src/Http/Routing/src/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -133,7 +133,7 @@ public Task ApplyAsync(HttpContext httpContext, CandidateSet candidates)
                 var httpMethod = httpContext.Request.Method;
                 var headers = httpContext.Request.Headers;
                 if (metadata.AcceptCorsPreflight &&
-                    string.Equals(httpMethod, PreflightHttpMethod, StringComparison.OrdinalIgnoreCase) &&
+                    (object.ReferenceEquals(httpMethod, PreflightHttpMethod) || StringComparer.OrdinalIgnoreCase.Equals(httpMethod, PreflightHttpMethod)) &&
```

Comment:
This also looks like an API we could add to HttpMethods. `bool HttpMethods.Equals(string, string)`